indico/modules/events/persons/controllers.py

```diff
@@ -0,0 +1,121 @@
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import itertools
+from collections import defaultdict
+
+from sqlalchemy.orm import noload
+
+from indico.core.db.sqlalchemy.principals import EmailPrincipal, PrincipalType
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.contributions.models.persons import ContributionPersonLink, SubContributionPersonLink
+from indico.modules.events.contributions.models.principals import ContributionPrincipal
+from indico.modules.events.contributions.models.subcontributions import SubContribution
+from indico.modules.events.models.persons import EventPerson
+from indico.modules.events.models.principals import EventPrincipal
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.modules.events.sessions.models.persons import SessionBlockPersonLink
+from indico.modules.events.sessions.models.principals import SessionPrincipal
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.events.persons.views import WPManagePersons, WPManagePendingPersons
+from indico.web.flask.util import url_for
+from MaKaC.webinterface.rh.conferenceModif import RHConferenceModifBase
+
+
+class RHPersonsList(RHConferenceModifBase):
+
+    CSRF_ENABLED = True
+
+    def _process(self):
+        persons_links = set(ContributionPersonLink.query.join(Contribution)
+                            .filter(Contribution.event_id == self.event_new.id,
+                                    ContributionPersonLink.is_speaker)
+                            .all())
+        persons_links.update(itertools.chain(*[s.conveners for s in self.event_new.sessions]))
+        persons = defaultdict(lambda: {'convener': False, 'speaker': False, 'contributions': defaultdict(dict),
+                                       'session_blocks': defaultdict(dict)})
+        for link in persons_links:
+            if isinstance(link.person.principal, EmailPrincipal):
+                continue
+            data = persons[link.person]
+            if isinstance(link, ContributionPersonLink):
+                data['speaker'] = True
+                person_contribs = (Contribution.query.with_parent(self.event_new)
+                                   .options(noload('*'))
```

Comment:
i'd rather be explicit instead of noloading everything. there's not much that's eager-loaded anyway